department-of-veterans-affairs/caseflow
# Enable sending inbox messages during job note creation

Connects #12524 

### Description
This enables the "Send as message to user" checkbox in the job note creation form. New messages will end up in the Caseflow Inbox of the job's asyncable user (its current "owner").

### Acceptance Criteria
- [ ] Adding a note with `send_to_intake_user: true` sends a message to the user's inbox
- [ ] Adding a note with `send_to_intake_user: false` doesn't send any message

### Testing Plan
1. Go to `/jobs` with at least one async job created.
2. Click into a job.
3. Write a job note, make sure the checkbox is selected, and click "Add Note"
4. As the owner of that job (possibly a different user), go to `/inbox`.
5. Verify that a new inbox message was created from that job note.
6. Click on the link in the message to verify that it goes back to the job note.

### User Facing Changes

This checkbox, originally made by @pkarman, has been uncommented back in:
![Screen Shot 2019-11-13 at 17 14 20](https://user-images.githubusercontent.com/282869/68810541-5c9fc380-063c-11ea-987d-4e3087e5657c.png)

Caseflow Inbox has already been built, but this is the first time it will be populated with actual messages. A job-note message would look like:
<img width="1085" alt="Screen Shot 2019-11-17 at 12 11 56" src="https://user-images.githubusercontent.com/282869/69011006-c3bac200-0933-11ea-9744-9458fe875927.png">
 
### Implementation Notes

- The Message object has a polymorphic detail field, currently unused. For job note messages, I'm passing the job note itself, which feels like the most specific and informative option, should we need it one day.
- Changed `Asyncable#asyncable_user` to return a User object rather just the `css_id`, which is more surprising and less useful in most situations.
- Changed the checkbox to be disabled only during `saveInProgress`, not when the text field is empty. An empty text field should only disable the "Add Note" button.
